### PR TITLE
Index operation

### PR DESCRIPTION
**Pull request by xluo-aws**
### Description
[Add option to open index, set index status to block write on split page to make the index ready for split]

### Issues Resolved
[[List any issues this PR will resolve](https://github.com/opensearch-project/index-management-dashboards-plugin/issues/361)]

### Check List
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/pages/Indices/containers/IndicesActions/index.tsx [hunk 1]
```diff
@@ -346,6 +369,8 @@ export default function IndicesActions(props: IndicesActionsProps) {
           onSplitIndex={splitIndex}
           getIndexSettings={getIndexSettings}
           coreServices={coreServices}
```
Comment (SuZhou-Joe) [inline public/pages/Indices/containers/IndicesActions/index.tsx:371]: We can remove the coreServices here.

**Review by SuZhou-Joe — DISMISSED** [review]

File: package.json [hunk 2]
```diff
@@ -63,7 +63,7 @@
     "eslint-plugin-no-unsanitized": "^3.0.2",
     "eslint-plugin-prefer-object-spread": "^1.2.1",
     "flat": "^4.1.1",
-    "husky": "^4.2.5",
+    "husky": "^3.0.0",
     "lint-staged": "^10.2.0",
```
Comment (Hailong-am) [inline package.json:67]: how about lint-staged, shall we revert to ` "lint-staged": "^9.2.0",`

File: public/pages/Indices/components/SplitIndexFlyout/SplitIndexFlyout.tsx [hunk 3]
```diff
@@ -89,8 +64,56 @@ export default class SplitIndexFlyout extends Component<SplitIndexProps> {
...
+    const blocksWriteValue = get(sourceSettings, ["index.blocks.write"]);
+
+    if (sourceIndex.health === "red") {
+      reasons.push(<>Source index health must not be red.</>);
+    }
+
+    if (sourceIndex.status === "close") {
+      reasons.push(
+        <>
+          Source index must not be in close status. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
```
Comment (Hailong-am) [inline public/pages/Indices/components/SplitIndexFlyout/SplitIndexFlyout.tsx:82]: could we use `{"    "}` to replace `&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`

File: public/pages/Indices/components/SplitIndexFlyout/SplitIndexFlyout.tsx [hunk 4]
```diff
@@ -89,8 +64,56 @@ export default class SplitIndexFlyout extends Component<SplitIndexProps> {
...
+        </>
+      );
+    }
+
+    if (sourceSettings && (!blocksWriteValue || blocksWriteValue !== "true")) {
+      const flat = true;
+      const blocksWriteSetting = { "index.blocks.write": "true" };
+      reasons.push(
+        <>
+          Source index must be in block write status. &nbsp;&nbsp;
```
Comment (Hailong-am) [inline public/pages/Indices/components/SplitIndexFlyout/SplitIndexFlyout.tsx:102]: could we use {" "} to replace `&nbsp;`

File: public/pages/Indices/components/SplitIndexFlyout/SplitIndexFlyout.tsx [hunk 5]
```diff
@@ -143,82 +168,104 @@ export default class SplitIndexFlyout extends Component<SplitIndexProps> {
...
+            "data-test-subj": "Number of shards",
+            min: Number(sourceIndex.pri) * 2,
+            step: Number(sourceIndex.pri),
+          },
         },
       },
     ];
 
+    if (reasons.length > 0) {
+      console.log(
```
Comment (Hailong-am) [inline public/pages/Indices/components/SplitIndexFlyout/SplitIndexFlyout.tsx:181]: pls remove `console.log`

**Review by SuZhou-Joe — APPROVED** [review]

**Review by Hailong-am — APPROVED** [review]

**Pull request merged**